packages/react/src/components/createInlineOverlayComponent.tsx

```diff
@@ -99,7 +114,36 @@ export const createInlineOverlayComponent = <PropType, ElementType>(
 
     componentDidUpdate(prevProps: IonicReactInternalProps<PropType>) {
       const node = this.ref.current! as HTMLElement;
-      attachProps(node, this.props, prevProps);
+
+      const newProps = {
+        ...this.props,
+        delegate: this.delegate,
+      };
+
+      attachProps(
+        node,
+        newProps,
+        prevProps
+      );
+    }
+
+    shouldComponentUpdate(nextProps: IonicReactInternalProps<PropType>) {
+      // Check if the overlay component is about to dismiss
+      if (
+        this.ref.current &&
+        nextProps.isOpen !== this.props.isOpen &&
+        nextProps.isOpen === false
+      ) {
+        this.isDismissing = true;
+      }
+
+      return true;
+    }
+
+    componentWillUnmount() {
+      if (this.ref.current) {
+        this.ref.current.dismiss();
+      }
```

Comment:
While playing around in a small test app, I noticed some errors when doing replace or push operations that I think are due to the timing of this dismiss call.

Repo/branch: https://github.com/amandaejohnston/sandbox-ionic-react/tree/FW-1889
Screencast: https://user-images.githubusercontent.com/90629384/194619453-2bab84cd-bfa6-421f-a9dd-ea68dec8b36c.mp4

My guess is this is happening because the `dismiss` event is firing and/or being picked up by the `Home` component *after* the modal has been removed from the DOM. This would cause the `onDidDismiss` handler for the modal to try and set the `isOpen` state on the modal when it's unmounted, throwing the error. Although, the fact that I don't get the error when doing a goBack operation may throw a wrench in that. 🤔 

Let me know if this is an implementation issue on my end, of course! The code is mostly copy-pasted from the test app page in this PR but maybe something went screwy in the process.